### PR TITLE
feat(switch): new design

### PR DESCRIPTION
**Pull request by rachelbt**
Removed the success connotation option.

image

### CODE REVIEW

**Review by YonatanKra — DISMISSED** [review]
LGTM

**Review by YonatanKra — COMMENTED** [review]
Tests are failing. What tests do youbthink to add?

File: components/switch/src/vwc-switch.scss [hunk 1]
```diff
@@ -1,138 +1,165 @@
 @use 'sass:selector';
 @use '@vonage/vvd-design-tokens/build/scss/semantic-variables/scheme-variables';
 @use '@vonage/vvd-foundation/scss/mixins/connotation/config' with (
+//'Connotation.Success' is  *deprecated*  and will be removed in vivid-3. Use any of the other connotation:
```
Comment (yinonov) [inline components/switch/src/vwc-switch.scss:4]:
```suggestion
//'Connotation.Success' is  *deprecated*  and will be removed in vivid-3. Use any of the other connotation predefined values:
```

File: components/switch/src/vwc-switch.scss [hunk 2]
```diff
@@ -1,138 +1,165 @@
...
-			transform: translateX(-18px);
-		}
+.mdc-switch__thumb-underlay {
+	top: 0;
+	left: 0;
+	width: 20px;
+	height: 100%;
+
+	.mdc-switch__thumb {
+		--nobe-size: var(--basic-nobe-size);
```
Comment (yinonov) [inline components/switch/src/vwc-switch.scss:135]: knob?

File: components/switch/src/vwc-switch.scss [hunk 3]
```diff
@@ -1,138 +1,165 @@
...
-			border: 0 none;
+			--basic-nobe-size: 14px;
+			background-color: var(--surface);
 		}
-	}
-
-	.mdc-switch--checked {
 		.mdc-switch__thumb-underlay {
-			transform: translateX(10px);
+			transform: translateX(var(--translate));
```
Comment (yinonov) [inline components/switch/src/vwc-switch.scss:38]: just a side note - use css variables with caution as they can lead to abuse if not handled correctly. I mean if there are states where they’re not defined by inner implementation, that’s a pit fall where piercing variables can damage the style. `translate` is a very generic variable name

**Review by yinonov — CHANGES_REQUESTED** [review]

**Comment by rachelbt** [comment]
> Tests are failing. What tests do youbthink to add?

Now all is OK

File: components/switch/src/vwc-switch.scss (hunk 3 above)
Comment (rachelbt) [reply components/switch/src/vwc-switch.scss]: changed the variables names :)

File: components/switch/src/vwc-switch.scss [hunk 4]
```diff
@@ -9,11 +9,11 @@
...
 	//css variables
-	--basic-nobe-size: 12px;
-	--translate: 16px;
-	--translate-natural: -16px;
+	--basic-knob-size: 12px;
+	--knob-translate: 16px;
+	--knob-translate-natural: -16px;
 	--natural-50: #757575;
-	--surface: #f2f2f2;
+	--switch-surface: #f2f2f2;
```
Comment (yinonov) [inline components/switch/src/vwc-switch.scss:16]: should be kept as constants. in a different variables partial file.

File: components/switch/src/vwc-switch.scss [hunk 5]
```diff
@@ -9,11 +9,11 @@
 
 :host {
 	//css variables
-	--basic-nobe-size: 12px;
-	--translate: 16px;
-	--translate-natural: -16px;
+	--basic-knob-size: 12px;
+	--knob-translate: 16px;
+	--knob-translate-natural: -16px;
 	--natural-50: #757575;
```
Comment (yinonov) [inline components/switch/src/vwc-switch.scss:15]:
```suggestion
	--knob-canvas: #757575;
```

**Review by yinonov — CHANGES_REQUESTED** [review]

File: components/switch/src/vwc-switch.scss (hunk 5 above)
Comment (rachelbt) [reply components/switch/src/vwc-switch.scss:15]: its not the knob canvas thought. Its the outline and the know when not active

Comment (yinonov) [reply components/switch/src/vwc-switch.scss:15]: so this?
```suggestion
	--knob-idle: #757575;
```

**Review by yinonov — CHANGES_REQUESTED** [review]

File: components/switch/src/vwc-switch.scss (hunk 5 above)
Comment (rachelbt) [reply components/switch/src/vwc-switch.scss]: Replaced it with `switch-default-color`.
Moved the colors to switch-variables but kept the other css variable fo dynamic changes

File: components/switch/src/vwc-switch.scss [hunk 6]
```diff
@@ -1,138 +1,168 @@
...
 	$default: primary,
 );
 @use '@vonage/vvd-foundation/scss/mixins/connotation' as connotation;
+@use 'partials/vwc-switch-variables' as switch-variables;
+
+:host {
+	// starting variables
+	#{switch-variables.$knob-size}: 12px;
+	#{switch-variables.$knob-translate}: 16px;
+	#{switch-variables.$knob-translate-natural}: -16px;
```
Comment (yinonov) [inline components/switch/src/vwc-switch.scss:15]: still here

File: components/switch/src/partials/_vwc-switch-variables.scss [hunk 7]
```diff
@@ -0,0 +1,8 @@
+//switch colors
+$switch-default-color: #757575;
+$switch-surface: #f2f2f2;
+
+//switch sizes
+$knob-translate: --knob-translate;
+$knob-translate-natural: --knob-translate-natural;
```
Comment (yinonov) [inline components/switch/src/partials/_vwc-switch-variables.scss:7]: still here

**Review by yinonov — CHANGES_REQUESTED** [review]

File: components/switch/src/partials/_vwc-switch-variables.scss (hunk 7 above)
Comment (rachelbt) [reply components/switch/src/partials/_vwc-switch-variables.scss:7]: I'll remove it 😵‍💫

**Review by yinonov — APPROVED** [review]

**Pull request merged**